working/wildcards/feature-specification.md

```diff
@@ -395,8 +395,18 @@ However, this *is* a breaking change. If this ships in the same version as
 pattern matching, we can gate it behind a language version and only break code
 when it upgrades to that version.
 
+### Existing Lints
+
+We have an existing [`no_wildcard_variable_uses`](https://dart.dev/tools/linter-rules/no_wildcard_variable_uses) lint, which advises users to avoid using wildcard parameters or variables.
+
+This lint is included in the core lint set which means that the scale of the breaking change should be small since most projects should have this lint enabled.
```

Comment:
I guess identifiers like `___` are used exclusively as homegrown wildcards today. Otherwise:

This lint will flag usages of names of the form `__`, `___` and so on, in addition to `_`. Does it give rise to unnecessary churn to flag all the names that have more than one underscore, given that they will keep their status as regular (non-wildcard) identifiers? For example, if someone really wants to use local variables with the name `_` then they might switch to `__` just because it's similar, and not overly long, and they might be unhappy if they still get the lint message.

At least, the lint could make the distinction, or we might want a separate lint that only targets the cases where `_` resolves to a declaration which would be wildcarded when that feature is enabled.

Reply:
Users would be avoiding all types of `_` which may be unnecessary churn for the names with more than one underscore, but if they're already using it and the lint is out there in the core set of lints.

I'm not sure if there's enough value to go back and change the lint now that it's already over-reporting. It's also a set of migrations that regardless, the code author has to go in and change once they learn about the wildcard feature.

And it would be good to have a separate lint that tells users that `_` will be wildcarded once the feature is enabled, I'm just not sure how that would interact with the `no_wildcard_variable_uses` lint. Would these two overlap if they were both in the core lint set?

@pq Maybe you'd have a better sense of these two lints and their use cases.

Reply:
> Does it give rise to unnecessary churn to flag all the names that have more than one underscore, given that they will keep their status as regular (non-wildcard) identifiers?

Great question. I'm really not sure! I'd be curious to know how often this happens in practice... I'm guessing not terrifically often but it'd be interesting to do a little grepping.

> Maybe you'd have a better sense of these two lints and their use cases.

I'd have to think about this a bit more... How about we open an issue on the linter repo and take the conversation there? I think this is safe to land as is but wouldn't want @eernstg's considerations to just get lost.

Reply:
Filed https://github.com/dart-lang/linter/issues/4968.
Let's continue the conversation there so we don't lose it.
I'll submit this.